hoomd/md/EvaluatorPairALJ.h

```diff
@@ -0,0 +1,1424 @@
+        }
+
+        {
+        Scalar x_two_z = q.v.x * two_z;
+        Scalar s_two_y = q.s * two_y;
+        mat[0][2] = x_two_z + s_two_y;
+        mat[2][0] = x_two_z - s_two_y;
+        }
+    }
+
+//! Anisotropic LJ (ALJ) potential.
+/*! The ALJ potential is a generalization of Lennard-Jones potential for convex
+ * anisotropic shapes. The potential is defined by a mean-field approximation
+ * to the total interaction energy between a pair of anistropic shapes that
+ * reduces the total interaction to a simple sum between 1) a contact
+ * interaction at between the closest points on each shape to the other, and 2)
+ * a center-center interaction analogous to the standard isotropic LJ
+ * potential.
+ */
+template<unsigned int ndim> class EvaluatorPairALJ
+    {
+    public:
+    //! Potential parameters for the ALJ potential.
+    struct param_type
+        {
+        HOSTDEVICE param_type()
+            : epsilon(0.0), sigma_i(0.0), sigma_j(0.0), contact_sigma_i(0.0), contact_sigma_j(0.0),
+              alpha(0), average_simplices(false)
+            {
+            }
+
+        //! Load dynamic data members into shared memory and increase pointer
+        /*! \param ptr Pointer to load data to (will be incremented)
+         *  \param available_bytes Size of remaining shared memory allocation
+         */
+        DEVICE void load_shared(char*& ptr, unsigned int& available_bytes) { }
+
+        DEVICE void allocate_shared(char*& ptr, unsigned int& available_bytes) const { }
+
+#ifndef __HIPCC__
+        //! Shape constructor
+        param_type(Scalar _epsilon,
+                   Scalar _sigma_i,
+                   Scalar _sigma_j,
+                   Scalar _contact_sigma_i,
+                   Scalar _contact_sigma_j,
+                   unsigned int _alpha,
+                   bool _average_simplices,
+                   bool managed)
+            : epsilon(_epsilon), sigma_i(_sigma_i), sigma_j(_sigma_j),
+              contact_sigma_i(_contact_sigma_i), contact_sigma_j(_contact_sigma_j), alpha(_alpha),
+              average_simplices(_average_simplices)
+            {
+            }
+
+        param_type(pybind11::dict params, bool managed)
+            : epsilon(params["epsilon"].cast<Scalar>()), sigma_i(params["sigma_i"].cast<Scalar>()),
+              sigma_j(params["sigma_j"].cast<Scalar>()),
+              contact_sigma_i((params["sigma_i"] * params["contact_ratio_i"]).cast<Scalar>()),
+              contact_sigma_j((params["sigma_j"] * params["contact_ratio_j"]).cast<Scalar>()),
+              alpha(params["alpha"].cast<unsigned int>()),
+              average_simplices(params["average_simplices"].cast<bool>())
+            {
+            }
+
+        pybind11::object toPython()
+            {
+            using namespace pybind11::literals;
+            return pybind11::dict("epsilon"_a = epsilon,
+                                  "sigma_i"_a = sigma_i,
+                                  "sigma_j"_a = sigma_j,
```

Comment:
What is this `_a` syntax?

Reply:
It lets you create a dictionary all at once using a Python like syntax. It is a C++ literal that lets you create a string key for a dictionary.